src/rust/runtime/memory/buffer/demibuffer.rs

```diff
@@ -617,9 +678,20 @@ impl Clone for DemiBuffer {
                         clone.buf_len = original.buf_len;
                         clone.data_off = original.data_off;
                         clone.nb_segs = original.nb_segs;
-                        clone.ol_flags = original.ol_flags | METADATA_F_INDIRECT; // Add indirect flag to clone.
                         clone.pkt_len = original.pkt_len;
                         clone.data_len = original.data_len;
+
+                        // Special case for zero-length buffers.
+                        if original.buf_len == 0 {
+                            debug_assert_eq!(clone.buf_len, 0);
+                            debug_assert_eq!(clone.buf_addr, ptr::null_mut());
```

Comment:
These are not in the critical-patch, so why should we use a debug_assert?

Reply:
I don't think we know in general when someone calls `clone()` on a buffer whether or not it is in the critical-path.  Offhand, I'd think most of the cases where we could call `clone()` are in the critical-path.  For the specific case here, which only applies to cloning a zero-length buffer, there is only one place (I think) currently in the system where we do that -- when sending a TCP FIN.  I think I'd consider sending a FIN to be more-or-less as critical-path as sending actual data, so I think it makes sense to leave these as `debug_assert`s.  We also don't know if someone might add another clone call or zero-length buffer in the future that would hit the combination of the two in the critical-path.

Reply:
ok.